src/api/l_math_vectors.c

```diff
@@ -517,7 +545,10 @@ static int l_lovrVec3Add(lua_State* L) {
   vec3 v = luax_checkvector(L, 1, V_VEC3, NULL);
   if (lua_type(L, 2) == LUA_TNUMBER) {
     float x = lua_tonumber(L, 2);
-    v[0] += x, v[1] += x, v[2] += x, v[3] += x;
+    v[0] += x;
+    v[1] += luax_optfloat(L, 3, x);
+    v[2] += luax_optfloat(L, 4, x);
+    v[3] += luax_optfloat(L, 5, x);
```

Comment:
Even though the 4th component is unused, for some reason it doesn't feel right to read a value for it (talking to Lua when we don't need to?).  I'd rather add zero or x to it.  Same for `sub`.

Reply:
Good point!